bootstrap/eks/api/v1beta2/nodeadmconfig_types.go

```diff
@@ -0,0 +1,195 @@
+package v1beta2
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+)
+
+// NodeadmConfigSpec defines the desired state of NodeadmConfig.
+type NodeadmConfigSpec struct {
+	// Kubelet contains options for kubelet.
+	// +optional
+	Kubelet *KubeletOptions `json:"kubelet,omitempty"`
+
+	// Containerd contains options for containerd.
+	// +optional
+	Containerd *ContainerdOptions `json:"containerd,omitempty"`
+
+	// Instance contains options for the node's operating system and devices.
+	// +optional
+	Instance *InstanceOptions `json:"instance,omitempty"`
+
+	// FeatureGates holds key-value pairs to enable or disable application features.
+	// +optional
+	FeatureGates map[Feature]bool `json:"featureGates,omitempty"`
+
+	// PreBootstrapCommands specifies extra commands to run before bootstrapping nodes.
+	// +optional
+	PreBootstrapCommands []string `json:"preBootstrapCommands,omitempty"`
```

Comment:
wdyt calling it `PreNodeadmCommands` to match with `NodeConfig` and `PreKubeadmCommands` ?